libsolidity/codegen/CompilerContext.cpp

```diff
@@ -167,6 +167,18 @@ CompilerContext& CompilerContext::appendJump(eth::AssemblyItem::JumpType _jumpTy
 	return *this << item;
 }
 
+CompilerContext& CompilerContext::appendInvalid()
+{
+	return *this << Instruction::INVALID;
+}
+
+CompilerContext& CompilerContext::appendConditionalInvalid()
+{
+	eth::AssemblyItem falseTag = appendConditionalJump();
```

Comment:
I intentionally tried not to be too clever about encoding the jumps. I had considered the same optimization. I have pushed this now.